data/tasks.js

```diff
@@ -112,37 +141,54 @@ let exportedMethods = {
                  };
                   
                  let updateInfo = await tasksCollection.updateOne(
-                   { _id: id },
+                   { _id: Id },
                    { $set: updated }
                  );
-           return {"taskID": id, "completed": true}
+
+                 if (updateInfo.modifiedCount !== 1) {
+                  throwError
+                    "Could not update"
+                
+                }
+
+           return {"taskID": Id, "completed": true}
           },
 
+    async create(userId , name, importance, deadlineDate) {
+    //  console.log("here")
+      userId = validatetId(xss(userId)); 
+
+      const targetUser = await userData.get(userId);
+    
+    name=name.trim()
 
-    async create(name, importance, deadlineDate) {
-      if(!verify.validString(name)){
```

Comment:
I think you might have deleted the string verification for `name`, please re-add it